### ISSUE TITLE
t/run/locale.t fails with Perl 5.37.11 on Fedora

### ISSUE THREAD

**Issue by jplesnik** [Perl/perl5#21113]
I am preparing update of Perl in Fedora Rawhide and I use perl-5.37.11 for the test. 
When I ran build I found the test `t/run/locale.t` was failing:

```
# Failed test 40 - In setting complicated valid LC_ALL, final individ category doesn't need a ';' at run/locale.t line 534
#      got "\nThe following characters (and maybe others) may not have the same meaning as the Perl program expects: I i\n; codeset=ISO-8859-9"
# expected ""
#  diff at 0
#    after "\nThe following characters (and maybe oth"
#     have "\nThe following characters (and maybe oth"
#     want ""
# PROG: 
#                 use locale;
#                 use POSIX;
#                 POSIX::setlocale(LC_ALL, "LC_COLLATE=tr_CY;LC_CTYPE=tr_CY;LC_MESSAGES=tr_CY;LC_MONETARY=tr_CY;LC_NUMERIC=tr_CY;LC_TIME=tr_CY;LC_ADDRESS=tr_CY;LC_IDENTIFICATION=tr_CY;LC_MEASUREMENT=tr_CY;LC_PAPER=tr_CY;LC_TELEPHONE=tr_CY");
# STATUS: 0
sh: warning: setlocale: LC_ALL: cannot change locale (This is not a legal locale name): No such file or directory
run/locale.t ......................................................... 
Failed 1/43 subtests 
	(less 2 skipped subtests: 40 okay)
```

I executed the program with perl 5.36.1 and 5.37.11
```
use locale;
use POSIX;
POSIX::setlocale(LC_ALL, "LC_COLLATE=tr_CY;LC_CTYPE=tr_CY;LC_MESSAGES=tr_CY;LC_MONETARY=tr_CY;LC_NUMERIC=tr_CY;LC_TIME=tr_CY;LC_ADDRESS=tr_CY;LC_IDENTIFICATION=tr_CY;LC_MEASUREMENT=tr_CY;LC_PAPER=tr_CY;LC_TELEPHONE=tr_CY");
```
Output 5.36.1:
```
Locale 'tr_CY' may not work well.
The following characters (and maybe others) may not have the same meaning as the Perl program expects:
I i
; codeset=ISO-8859-9
```

Output 5.37.11:
```

The following characters (and maybe others) may not have the same meaning as the Perl program expects: I i
; codeset=ISO-8859-9
```

The change in output is caused by commit 667fa5f07f9 where `PL_warn_locale` was rewritten.
That is the reason why  https://github.com/Perl/perl5/blob/blead/t/loc_tools.pl#L182 does not work like before.

**Comment by jkeenan** [comment]
@jplesnik, can you test the patch suggested by @khwilliamson in https://github.com/Perl/perl5/pull/21124 and let us know if it addresses your problem?  Thanks.

**Comment by jplesnik** [comment]
@jkeenan , I've tested it and it solved the error. Thank you.

**Comment by khwilliamson** [comment]
I believe this should be a release blocker.

**Comment by khwilliamson** [comment]
I further researched this, and discovered it is a regression due to 667fa5f07f977e11018489f79c0e6531c9f79a72 in 5.37.  And that there is hence a better fix than was submitted and approved in #21124; I have updated that PR to be completely different. @jplesnik could you please test this revision; and sorry for the extra effort.

**Comment by jplesnik** [comment]
@khwilliamson I tried it and the test passed.
